### ISSUE TITLE
HELP to Translate

### ISSUE THREAD

**Issue by milesibastos** [dabbott/react-native-express#23]
Hi, congratulations for the great job!
I'm thinking about translate to pt-BR but I'm not sure the best approach to do it.
Maybe convert it to GitBook would be an option to help extract the markdown from the code an translate it...
I would like to hear you about it.

**Comment by dabbott** [comment]
Hi! Translations would be very helpful!

I think we need the ability to switch back and forth between markdown, React, and other JS code, so I don't think GitBook will work. (although I'm not sure - I haven't used GitBook)

I would probably create a separate directory, e.g. `Pages/pt-BR`, with the translated pages in that directory. We would move existing pages to `Pages/en-US`. Then depending on the user's language setting, we could load from the translated directory. (We would do that here: https://github.com/dabbott/react-native-express/blob/master/src/pages/createProxyComponent.js#L11)

So the components would be entirely separate, loaded on demand. This gives us a lot of flexibility. Any common code between languages (e.g. the examples, if there are no comments to translate) could then be pulled out into a separate shared directory.

**Comment by francescjimenez** [comment]
Hi! It's not complicated:
https://toolchain.gitbook.com/languages.html

I could help the translation to Spanish

**Comment by josenaves** [comment]
@dabbott  I can also help you to translate to pt-BR. Can I just fork this repo and do the translation?
Or are we going through something more sophisticated scheme ?

**Comment by josenaves** [comment]
@dabbott we want to translate the document. Can I make a fork and just translate it ?

**Comment by lowkeypapi** [comment]
@josenaves How are you planning to translate? any tools(gitbook) or approaches you're going to follow? Maybe we can discuss it here and whoever wants to translate will follow whatever guide we come with

**Comment by josenaves** [comment]
@lowkeypapi  In fact, I don't think I'd use Gitbook (see https://github.com/dabbott/react-native-express/issues/23#issuecomment-276236133).

What I'm intend to do is just create a fork and start coding. Then @dabbott could relocate the pt-BR version into a new directory and make the adjustments in the main English page.

**Comment by lowkeypapi** [comment]
@josenaves Sounds like a plan. Let the games begin

**Comment by dabbott** [comment]
:+1: sounds great, I like the fork plan. I can move the files around whenever you’re ready.

**Comment by josenaves** [comment]
Awesome ! 

I've just translated the introduction.

Quick question for you @dabbott :

Do you think I should translate the code (examples) also ?

**Comment by dabbott** [comment]
Hmmm, I don’t really know. I don’t have a strong opinion either way - whatever you want to do is good. I’m sort of under the impression that most code is in English even if the tutorial is in another language.

**Comment by josenaves** [comment]
Yes, @dabbott ! I feel the same about using English in code. But I cannot tell just by myself.

Well, I'll keep the code in English for while.

**Comment by cnscorpions** [comment]
Hey, you guys, I'd like to fork it and translate from English to Chinese :)

**Comment by dabbott** [comment]
That'd be awesome!

**Comment by cnscorpions** [comment]
@dabbott Translation to Chinese is in progress [here](https://github.com/cnscorpions/react-native-express-cn). Please add it in readme of this repo. 😆 thanks

**Comment by cnscorpions** [comment]
@dabbott got it. Thanks

**Comment by fegvilela** [comment]
What's the pt-BR translation status? Is it complete?
